pkg/object/mqttproxy/broker.go

```diff
@@ -450,6 +553,67 @@ func (b *Broker) sendMsgToClient(span *model.SpanContext, topic string, payload
 	}
 }
 
+func (b *Broker) processBrokerModePublish(clientID string, publish *packets.PublishPacket) {
```

Comment:
This function accomplishes two things: one is to publish a package to the subscribed devices on the current node, and the other is to publish a package to the subscribed devices on other nodes. I strongly recommend implementing this function as two separate functions and assembling them within this larger function.

Reply:
good advice, i will update it now